hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -449,6 +449,24 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		meta.SetStatusCondition(&hcluster.Status.Conditions, *condition)
 	}
 
+	// Copy the Degraded condition on the hostedcontrolplane
+	{
+		condition := &metav1.Condition{
+			Type:               string(hyperv1.HostedClusterDegraded),
+			Status:             metav1.ConditionUnknown,
+			Reason:             hyperv1.ClusterVersionStatusUnknownReason,
+			ObservedGeneration: hcluster.Generation,
+		}
+		if hcp != nil {
+			degradedCondition := meta.FindStatusCondition(hcp.Status.Conditions, string(hyperv1.HostedControlPlaneDegraded))
+			if degradedCondition != nil {
+				condition = degradedCondition
+				condition.Type = string(hyperv1.HostedClusterDegraded)
```

Comment:
would't we need to keep the `hcluster.Generation`? i.e in line 477 shouldn't we `condition.ObservedGeneration = hcluster.Generation` so it's always set?

Reply:
indeed :+1: